InstagramClone/View Controllers/HomeViewController.m

```diff
@@ -7,20 +7,111 @@
 
 #import "HomeViewController.h"
 #import "LogoutHandler.h"
+#import "ComposeViewController.h"
+#import "DetailsViewController.h"
 
-@interface HomeViewController ()
+#import "PostCell.h"
+#import "PostCellDecorator.h"
+
+#import "ParsePostHandler.h"
+
+@interface HomeViewController () <UITableViewDataSource, UITableViewDelegate, ComposeViewControllerDelegate, ParsePostHandlerDelegate>
+
+@property (weak, nonatomic) IBOutlet UITableView *timelineTableView;
+@property (nonatomic, strong) NSMutableArray<Post *> *posts;
+@property (nonatomic, strong) ParsePostHandler *postHandler;
 
 @end
 
 @implementation HomeViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    
+    self.timelineTableView.delegate = self;
+    self.timelineTableView.dataSource = self;
+    
+    self.postHandler = [[ParsePostHandler alloc] init];
+    self.postHandler.delegate = self;
+    [self fetchPosts];
+    
+    UIRefreshControl *refreshControl = [[UIRefreshControl alloc] init];
+    [refreshControl addTarget:self
+                       action:@selector(beginRefresh:)
+             forControlEvents:UIControlEventValueChanged];
+    [self.timelineTableView insertSubview:refreshControl atIndex:0];
+    
+    UINib *nib = [UINib nibWithNibName:@"PostCell" bundle:nil];
+    [self.timelineTableView registerNib:nib forCellReuseIdentifier:@"PostCellId"];
+}
+
+- (void)fetchPosts {
+    [self.postHandler queryHomePosts];
+}
+
+- (void)successfullyQueried:(NSMutableArray<Post *> *)posts {
+    self.posts = posts;
+    [self.timelineTableView reloadData];
+}
+
+- (void)postedSuccessfully {
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (void)failedToPost {
+    
 }
 
 - (IBAction)onTapLogout:(id)sender {
     LogoutHandler *logoutAction = [[LogoutHandler alloc] init];
     [logoutAction logout];
 }
 
+- (IBAction)onTapCompose:(id)sender {
+    UINavigationController *composeNavigationController = (UINavigationController*)[self.storyboard instantiateViewControllerWithIdentifier:@"ComposeNavigation"];
+   [self presentViewController:composeNavigationController animated:YES completion:nil];
```

Comment:
nit: I would present it after setting things on it. It's not gonna make a practical difference, it just makes more sense in my mind to configure then show.